docs/index.rst

```diff
@@ -46,6 +46,7 @@ write OCS Agents or Clients.
     developer/feeds
     developer/agents
     developer/clients
+    developer/timeout_lock
```

Comment:
I'm not sure this should get its own top level in the sidebar, I think it fits better under Agents. Though as I say that I realize the Agents page should probably be broken up a bit. I do think we should move this in there, whether now or later.